bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -338,6 +338,22 @@ class PlotCanvasView extends Renderer.View
             if is_panning?
               range_info['start'] = rng.get('start')
 
+    min_interval = rng.get('min_interval')
+    max_interval = rng.get('max_interval')
+    if min_interval? || max_interval?
+      new_interval = Math.abs(range_info['end'] - range_info['start'])
+      center = 0.5 * (rng.get('end') + rng.get('start'))
+      sign = if reversed then -1 else 1
+
+      if min_interval > 0
+        if new_interval < min_interval
+          range_info['start'] = center - sign * 0.5 * min_interval
+          range_info['end'] = center + sign * 0.5 * min_interval
+      if max_interval > 0
+        if new_interval > max_interval
+          range_info['start'] = center - sign * 0.5 * max_interval
+          range_info['end'] = center + sign * 0.5 * max_interval
```

Comment:
Good point. I think it should be relatively easy to fix though.

What's your take on having the axis being constrained individually? If you're fine about that, I'll make above change and this should be about done.